src/lib/utils/tests/test_validation.py

```
@@ -0,0 +1,245 @@
+            self.assertEqual(result, os.path.abspath(temp_dir))
+
+    def test_valid_path_with_nonexistent_path_raises_osmo_user_error(self):
+        with self.assertRaises(osmo_errors.OSMOUserError):
+            validation.valid_path('/nonexistent/path/that/does/not/exist/xyz123')
+
+
+class TestValidFilePath(unittest.TestCase):
+    """Tests for validation.valid_file_path."""
+
+    def test_valid_file_path_with_nonexistent_path_returns_path(self):
+        path = '/tmp/nonexistent_testbot_file_xyz_12345.txt'
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/nonexistent_testbot_file_xyz_12345.txt"

(S108)